src/main/java/com/xebialabs/deployit/ci/DeployitNotifier.java

```diff
@@ -364,7 +388,18 @@ public FormValidation doCheckApplication(@QueryParameter String credential, @Que
             return warning("Application does not exist, but will be created upon package import.");
         }
 
+        @POST
         public FormValidation doReloadTypes(@QueryParameter String credential, @AncestorInPath AbstractProject project) {
+            if (project == null) { // no context
+                if (!project.hasPermission(Jenkins.ADMINISTER)) {
+                    return FormValidation.ok();
+                }
+            } else if (!project.hasPermission(Item.EXTENDED_READ)
+                    && !project.hasPermission(CredentialsProvider.USE_ITEM)) {
+                return FormValidation.ok();
+            }
+            project.checkPermission(AbstractProject.CONFIGURE);
```

Comment:
As above, this seems wrong.

Reply:
removed unnecessary code and implemented form validation by checking permissions